src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -341,6 +349,27 @@ public override LookupResultKind ResultKind
                 return !this.OriginalIndexersOpt.IsDefault ? LookupResultKind.OverloadResolutionFailure : base.ResultKind;
             }
         }
+
+        public BoundIndexerAccess Update(AccessorKind accessorKind)
+        {
+            if (accessorKind == this.AccessorKind)
```

Comment:
This is probably redundant, the `Update` below is going to do the same comparison anyway.